tests/ui/significant_drop_tightening.rs

```diff
@@ -0,0 +1,16 @@
+#![warn(clippy::significant_drop_tightening)]
+
+use std::sync::Mutex;
+
+fn unnecessary_contention_with_multiple_owned_results() {
+    let mutex = Mutex::new(1i32);
+    let lock = mutex.lock().unwrap();
+    let rslt0 = lock.abs();
+    let rslt1 = lock.is_positive();
+    do_heavy_computation_that_takes_time((rslt0, rslt1));
+}
```

Comment:
It is still necessary to check cases with single a `lock` reference in order to suggest a single line merging like `mutex.lock().unwrap().is_positive()` but I would like to do so in a following PR if possible.